src/dmd/expressionsem.d

```diff
@@ -4011,8 +4032,45 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
         return exp.expressionSemantic(sc);
     }
 
+    void checkAccessSharedCall(CallExp exp)
+    {
+        assert (exp.f);
+        auto params = exp.f.getParameterList().parameters;
+        if (exp.arguments && exp.arguments.dim && params && params.dim)
+        {
+            foreach (i, arg;*exp.arguments)
+            {
+                if (arg.op == TOK.variable)
+                {
+                    auto ve = cast (VarExp) arg;
+                    auto vd = ve.var.isVarDeclaration();
+                    if (vd &&
+                        vd.type &&
+                        vd.type.isShared() &&
+                        i < params.dim &&
+                        !(*params)[i].type.isShared)
+                    {
+                        checkAccessShared(sc, arg.loc, vd);
+                    }
+                }
+            }
+        }
+    }
+
     override void visit(CallExp exp)
     {
+        // @@@SHARED@@@
+        // we need to set allow shared for overload resolution
```

Comment:
why should this affect overload resolution?

Reply:
why indeed :)
because for some reason overload-resolution needs to resolve properties.